server/src/main/java/org/elasticsearch/action/ActionListener.java

```diff
@@ -150,7 +151,7 @@ public String toString() {
      * the sense that an exception from the {@code onResponse} consumer is passed into the {@code onFailure} consumer.
      * <p>
      * If the {@code onFailure} argument is {@code listener::onFailure} for some other {@link ActionListener}, prefer to use
-     * {@link #delegateFailureAndWrap} instead.
+     * {@link #delegateFailureAndWrap} instead for performance reasons.
```

Comment:
https://github.com/elastic/elasticsearch/pull/96575 seems to have been about performance, but I can't easily discern the particulars to note here.

Reply:
I would guess it's (some subset of) avoiding the allocation of the method reference, reducing the layers of wrapping, and being easier for the compiler to spot and optimise the pattern.